sssom/parsers.py

```diff
@@ -612,26 +657,43 @@ def to_mapping_set_document(msdf: MappingSetDataFrame) -> MappingSetDocument:
         raise Exception("No valid prefix_map provided")
 
     mlist: List[Mapping] = []
-    ms = MappingSet()
+    if msdf.metadata is not None:
+        ms = MappingSet(
+            mapping_set_id=msdf.metadata["mapping_set_id"],
+            license=msdf.metadata["license"],
+        )
+    else:
+        ms = MappingSet(mapping_set_id=DEFAULT_MAPPING_SET_ID, license=DEFAULT_LICENSE)
     bad_attrs = {}
     if msdf.df is not None:
         for _, row in msdf.df.iterrows():
             mdict = {}
             for k, v in row.items():
-                ok = False
-                if k:
-                    k = str(k)
-                if hasattr(Mapping, k):
-                    mdict[k] = v
-                    ok = True
-                if hasattr(MappingSet, k):
-                    ms[k] = v
-                    ok = True
-                if not ok:
-                    if k not in bad_attrs:
-                        bad_attrs[k] = 1
-                    else:
-                        bad_attrs[k] += 1
+                # Check if v exists AND v is not nan
+                if v and v == v:
```

Comment:
Can we somehow refactor these lines with L200 ff? They seem to be doing mostly the same..

Reply:
L673-700, or even more.

Reply:
Done